source/fuzz/transformation_changing_memory_semantics.h

```diff
@@ -0,0 +1,118 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef SOURCE_FUZZ_TRANSFORMATION_CHANGING_MEMORY_SEMANTICS_H_
+#define SOURCE_FUZZ_TRANSFORMATION_CHANGING_MEMORY_SEMANTICS_H_
+
+#include "source/fuzz/protobufs/spirvfuzz_protobufs.h"
+#include "source/fuzz/transformation.h"
+#include "source/fuzz/transformation_context.h"
+#include "source/opt/ir_context.h"
+
+namespace spvtools {
+namespace fuzz {
+
+// This transformation is responsible for changing the mask of memory
+// semantics with a range of 5 bits of values.
+class TransformationChangingMemorySemantics : public Transformation {
+ public:
+  explicit TransformationChangingMemorySemantics(
+      protobufs::TransformationChangingMemorySemantics message);
+
+  TransformationChangingMemorySemantics(
+      const protobufs::InstructionDescriptor& atomic_instruction,
+      uint32_t memory_semantics_operand_position,
+      uint32_t memory_semantics_new_value_id);
+
+  // - |message_.atomic_instruction| atomic instruction that would like to
+  //   change its memory semantics value.
+  // - |message_.memory_semantics_operand_position| position of atomic
+  // instruction
+  //   would like to change, must be equal to 2 or 3 only.
+  // - |message_.memory_semantics_new_value_id| the new id of memory semantics
+  //   that is will change with the old.
+  bool IsApplicable(
+      opt::IRContext* ir_context,
+      const TransformationContext& transformation_context) const override;
+
+  // Change id of memory semantics for specific position with a new one.
+  void Apply(opt::IRContext* ir_context,
+             TransformationContext* transformation_context) const override;
+
+  // Check if instruction is atomic instruction only. Check if Memory semantics
+  // operand Position equal to 0 or 1. Check if operand position equal to 0 in
+  // the case of instruction that takes one memory semantics operand and 0 or 1
+  // for instructions that takes two operands.
+  static bool IsNeededOpcodeWithAppropriatePosition(SpvOp opcode,
+                                                    uint32_t position);
+
+  // - |opcode| opcode must be atomic or barrier instruction.
+  //   Valid memory semantics masks:
+  //   OpAtomicLoad: None | Acquire | SequentiallyConsistent
+  //   OpAtomicStore: None | Release | SequentiallyConsistent
+  //   OpAtomicRMW instructions: None | AcquireRelease | SequentiallyConsistent
+  //   Barrier instructions:  None | Acquire | Release | AcquireRelease |
+  //   SequentiallyConsistent
+  // - |lower_bits_old_memory_semantics| must be suitable for specific atomic
+  //   or barrier instruction.
+  // - |lower_bits_new_memory_semantics| should be larger than the old one,
+  //   must be suitable for specific atomic or barrier instruction.
+  // - |memory_model| return false if the memory model is Vulkan and memory
+  //   semantics is Sequentially Consistent.
```

Comment:
Thank you for writing this, but I suggest that, in this case, we leave the details within a comment in the function body (which I have already provided in another comment). I suggest that you replace this documentation comment with:

Returns true if the proposed memory order strengthening from |first_5bits_old_memory_semantics| to |first_5bits_new_memory_semantics| is both valid (according to the SPIR-V specification) and is actually a strengthening that would not change the semantics of a SPIR-V module. For full details, see the comments within the function body.

Reply:
Done.